src/Cli/dotnet/commands/dotnet-tool/install/ParseResultExtension.cs

```diff
@@ -61,7 +61,7 @@ public static VersionRange GetVersionRange(this ParseResult parseResult)
         private static bool IsBarePackageVersion (string packageVersion)
         {
             // Define a regular expression pattern to match the "a.b.c" format
-            string pattern = @"^\d+\.\d+\.\d+$";
+            string pattern = @"^\d+\.\d+\.\d+(-[\w\d]+(\.[\w\d]+)*)?(\+[\w\d]+(\.[\w\d]+)*)?$";
```

Comment:
I had perhaps a more straightforward idea - we have access to NuGet's [SemanticVersion](https://github.com/NuGet/NuGet.Client/blob/8bc8c9a2d55b13d033b14482c2beb4722f636c5b/src/NuGet.Core/NuGet.Versioning/SemanticVersion.cs#L16) class - could we use [`SemanticVersionFactory.TryParse`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Versioning/SemanticVersionFactory.cs#L32C31-L32C32) to handle the parsing instead of binding to a Regex?

Reply:
Thank you! It's updated